server/routes/requests.ts

```diff
@@ -37,336 +42,145 @@ router.get('/requests', async (req: Request, res: Response, next: NextFunction)
-                res.status(503).send({ message: 'Server error, try again later' });
-            }
-            next(err);
+        // does this need to test for requestAll?
+        if (status === 202 && !approved && accepted) {
+            await request.update({ id: requestId, approved: true });
+            postNotification(requestId, task.userId, 'approved', message, req.headers.authorization);
         }
-    },
-);
+        return status === 202 ? res.status(status).json(request) : res.status(status).send({ message });
+    } catch (err) {
+        next(err);
+        return res.status(503).send({ message: 'Server error, try again later' });
+    }
+});
 
-router.delete(
-    '/requests/:requestId',
-    async (req: Request, res: Response, next: NextFunction) => {
-        try {
-            jwt.verify(req.headers.authorization, cert);
-            const swapRequest = await db.Request.findOne({
-                where: { id: req.params.requestId },
-                attributes: [
-                    'id',
-                    'requesteeUserId',
-                    'type',
-                    'accepted',
-                    'approved',
-                    'taskId',
-                ],
-            });
-            if (swapRequest) {
-                await swapRequest.destroy().then(function () {
-                    res.status(200).json(swapRequest);
-                });
-            } else {
-                res.status(404).send({ message: 'Swap request not found' });
-            }
-        } catch (err) {
-            if (err instanceof TokenExpiredError || err instanceof JsonWebTokenError) {
-                res.status(401).send({ message: 'Unauthorized' });
-            } else {
-                res.status(503).send({ message: 'Server error, try again later' });
-            }
-            next(err);
+router.patch('/requests/accept/:requestId', certify, async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const { authorization } = req.headers;
+        const acceptingUserId = jwt.decode(authorization, { json: true }).sub.split('|')[1];
+        const request = await db.Request.findOne({
+            where: { id: req.params.requestId },
+            attributes: ['id', 'requesteeUserId', 'type', 'accepted', 'approved', 'replace'],
+            include: [
+                {
+                    model: db.Task,
+                    as: 'task',
+                    attributes: ['id', 'userId'],
+                },
+            ],
+        });
+        const { accepted, approved, task, id, type } = request || {};
+        const [message, status] = determineMessageStatus(request, accepted, approved);
+        if (status === 202) {
+            // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+            type === 'requestAll'
+                ? await request.update({ id, accepted: true, requesteeUserId: acceptingUserId })
+                : await request.update({ id, accepted: true });
```

Comment:
If you prefer, you could replace the eslint disable and ternary with an if block: 
```suggestion
            if (type === 'requestAll') {
                await request.update({ id, accepted: true, requesteeUserId: acceptingUserId });
            } else {
                await request.update({ id, accepted: true });
            }
```

Reply:
I'm probably gonna disable the eslint. this might be a bit personal preference, but if else blocks just feel.. more cluttered with the curly braces and the ternary feels cleaner